### PR TITLE
Move Developers documentation section

### PR DESCRIPTION
**Pull request by Mariatta**
I copied the Developers sections from coala/documentation into this repo.
Once this is up and running, we can work on removing the Developers section from coala/documentation.

Let me know what you think.
Thanks 😃 

I also brought in the logo and custom css from docs.coala.io.

### Checklist

- [x] I have rebased properly. Please see [our tutorial on rebasing](http://coala.readthedocs.io/en/latest/Developers/Git_Basics.html#rebasing).
- [x] I have gone through the [commit guidelines](http://coala.readthedocs.io/en/latest/Developers/Writing_Good_Commits.html) and I've followed them.
- [x] I have followed the [guidelines for the commit shortlog](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#shortlog).
- [x] I have followed the [guidelines for the commit body](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#commit-body).
- [x] I have included the issue URL with the 'Fixes' keyword if the commit fixes a *real bug* and the 'Closes' keyword if it adds a feature or enhancement. For details, check the [issue reference section in our commit guidelines](http://docs.coala.io/en/latest/Developers/Writing_Good_Commits.html#issue-reference).
- [x] I have [run all the tests](http://coala.readthedocs.io/en/latest/Developers/Executing_Tests.html) and they all pass. If any CI (below) is not green, please fix them. If GitMate issues appear you will have to amend your commits, pushing new commits on top is not sufficient!

### Reviewers

- copy coala/documentation/Developers/* into coala/docs/Developers/*
- match the style of docs.coala.io by bringing in logo and custom.css

Closes https://github.com/coala/coala/issues/3133

### CODE REVIEW

**Comment by sils** [comment]
why the help section? That one is for users as well right? Shouldn't that stay?

**Comment by sils** [comment]
btw. thanks so much @Mariatta I'm so super happy to see something movinghere thanks to you!

**Comment by Mariatta** [comment]
No problem :)
Re: Help Section, just thought developers might need help too, maybe having the link here is handy

We can keep the help section in both?

**Comment by Mariatta** [comment]
But I can remove it from here if that's preferred.

**Comment by sils** [comment]
keeping things twice is bad. Really bad. It leads to things being maintained in one place, getting outdated in the other.

**Comment by Mariatta** [comment]
Good point. I'll work on removing the Help section from this PR.

File: docs/Help/FAQ.rst [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  virtualenv -p python3 ~/venvs/coala
+  . ~/venvs/coala/bin/activate
+  pip install -U pip
+  pip install coala-bears
+
+
+should do the job. You have to activate this virtualenv on every terminal
+session you want to use coala though (tip: add it to bashrc!)
+
+What are those things failing/passing on my Pull Request?
```
Comment (Mariatta) [inline docs/Help/FAQ.rst]: I'm thinking that this section of the FAQ better suited together with the Developer's guide (here) as opposed to the user docs.

What do you think about keeping this one here?
Perhaps create Developers/FAQ.rst ?

Comment (sils) [inline docs/Help/FAQ.rst]: we can create a devs FAQ later, lets do the move and create the FAQ in an own commit/pr?

Comment (Mariatta) [inline docs/Help/FAQ.rst]: Thanks. I removed the Help section.

File: docs/Developers/Writing_Tests.rst [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+    statements will be executed from your component when invoking your
+    test cases. A branch coverage of 100% is needed for any commit in
+    order to be pushed to master - please ask us on gitter if you need
+    help raising your coverage!
+
+
+    The branch coverage can be measured locally with the
+    ``py.test --cov`` command.
+
+    .. seealso
```
Comment (Mariatta) [inline docs/Developers/Writing_Tests.rst]: Done :) Thanks.

**Comment by sils** [comment]
let's see how this looks, shall we?

https://github.com/coala/coala/pull/3134/commits/a0f9cba3c10f4e937fe38e65a743a27a8fe96b8f ack

**Comment by sils** [comment]
@rultor merge

**Pull request merged**